server/internal/infrastructure/aws/taskrunner.go

```
@@ -0,0 +1,106 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
+	"github.com/reearth/reearth-cms/server/pkg/asset"
+	"github.com/reearth/reearth-cms/server/pkg/task"
+	"github.com/reearth/reearthx/log"
+	"github.com/reearth/reearthx/rerror"
+)
+
+type TaskRunner struct {
+	topicARN   string
+	webhookARN string
+	snsClient  *sns.Client
+}
+
+func NewTaskRunner(ctx context.Context, conf *TaskConfig) (gateway.TaskRunner, error) {
+	if conf.WebhookARN == "" || conf.TopicARN == "" {
+		return nil, errors.New("Missing configuration")
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	snsClient := sns.NewFromConfig(cfg)
+
+	return &TaskRunner{
+		webhookARN: conf.WebhookARN,
+		topicARN:   conf.TopicARN,
+		snsClient:  snsClient,
+	}, nil
+}
+
+// Run implements gateway.TaskRunner
+func (t *TaskRunner) Run(ctx context.Context, p task.Payload) error {
+	if p.Webhook == nil {
+		return t.runTaskReq(ctx, p)
+	}
+	return t.runWebhookReq(ctx, p)
+}
+
+func (t *TaskRunner) Retry(ctx context.Context, id string) error {
+	panic("not implemented")
```

it's better to return an error

---

Done!